### PR TITLE
feat: Link node with target and rel

### PR DESCRIPTION
**Pull request by andrewchmr**
Issue: https://github.com/facebook/lexical/issues/2671

Support `target` and `rel` for LinkNode

Ideas for the future:
- Add an option in the [playground](https://playground.lexical.dev/) if link should be opened in the new tab

### CODE REVIEW

File: packages/lexical-link/src/index.ts [hunk 1]
```diff
@@ -111,10 +137,28 @@ export class LinkNode extends ElementNode {
...
+  setTarget(target: string | undefined): void {
+    const writable = this.getWritable();
+    writable.__target = target;
+  }
+
+  getRel(): string | undefined {
+    return this.getLatest().__rel;
+  }
+
+  setRel(rel: string | undefined): void {
```
Comment (thegreatercurve) [inline packages/lexical-link/src/index.ts:153]: Can we name this to be something more explicit, like `setRelationship`?

**Comment by thegreatercurve** [comment]
LGTM @andrewchmr! It just seems to be breaking a few tests to do with the Autolink node.

**Comment by andrewchmr** [comment]
> LGTM @andrewchmr! It just seems to be breaking a few tests to do with the Autolink node.

I fixed tests with `Autolink`, but there are 2 failing tests that don't seem to be related to my changes 🤔

image

**Comment by thegreatercurve** [comment]
@andrewchmr This is weird, I can't actually seeing any tests failing in the CI, especially the unit tests. Can you try rebasing onto main and then force pushing an update to trigger another CI run? It might be some weird caching issue with Github Actions

**Comment by thegreatercurve** [comment]
The failing unit tests are due to a the coverage reports returning errors. Given that we don't use the coverage reports, we can probably just delete them: https://github.com/facebook/lexical/pull/2699

**Review by thegreatercurve — APPROVED** [review]
LGTM @andrewchmr! Thanks for making these backwards compatible and adding in the tests.

**Comment by thegreatercurve** [comment]
Closes #2671

**Comment by fameoflight** [comment]
Any ideas on how can I enable target="_blank" as default for any link

**Comment by giokallis** [comment]
Is there anything new with @fameoflight's proposal ? I also don't understand how to use the target in LinkNode!

**Comment by tylerklose** [comment]
> Is there anything new with @fameoflight's proposal ? I also don't understand how to use the target in LinkNode!

I tried achieving this by using the following, but have had no luck:
```javascript
  const URL_MATCHER =
    /((https?:\/\/(www\.)?)|(www\.))[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)/

  const MATCHERS = [
    (text: string) => {
      const match = URL_MATCHER.exec(text)
      return (
        match && {
          index: match.index,
          length: match[0].length,
          text: match[0],
          url: match[0],
          attributes: { rel: "noopener", target: "_blank" }
        }
      )
    },
  ]
  
  ...
  
  <AutoFocusPlugin />
  ```

**Pull request merged**